src/main/java/CharList.java

```diff
@@ -31,41 +34,38 @@ public int length() {
 
     public char charAt(int index) {
         return container.get(index);
-
     }
 
-    public int indexOf(char c) {
-        return container.indexOf(c);
 
+    public int indexOf(char character) {
+        return container.indexOf(character);
     }
 
     CharList subString(int start, int end) {
         return new CharList(new ArrayList<Character>(container.subList(start, end)));
-
     }
 
     CharList removeDuplicates() {
-        HashSet<Character> a = new HashSet<Character>();
-        a.addAll(container);
-        return new CharList(new ArrayList(a));
+        HashSet<Character> listOfCharacters = new HashSet<Character>(container);
+        return new CharList(new ArrayList(listOfCharacters));
     }
 
-    CharList removeFirst(Character c) {
-        container.remove(c);
+    CharList removeFirst(Character character) {
+        container.remove(character);
         return new CharList(new ArrayList<Character>(container));
```

Comment:
there is no point to create nu object on return, but if you want to keep your list intact, you need first of all to create new object with your list and after that make your changes